`$$` and `$` math syntax support

KSR-Yasuda opened the pull request:
Patch of [the #190 comment](https://github.com/previm/previm/issues/190#issuecomment-2874825546).
Make a PR anyway.

KSR-Yasuda committed:
`$$` and `$` math syntax support

KSR-Yasuda commented:
Perhaps it should have an option to disable the feature.

KSR-Yasuda commented:
Oh, this fails for `inline-codes` and

```
code blocks.
```

KSR-Yasuda committed:
Fix math expression replacement in code syntax

KSR-Yasuda commented:
Fixed to rollback math syntax replacement in code syntax.

After math expression replacement,
call `md.render()` and fix if they appear in `<pre>` or `<code>`.
To search `<pre>` and `<code>`, generate a dummy DOM object, and process its `innerHTML`.

#### Side effect

* Text `&#x24;` in code syntax is unnecessarily rolled back to `\$`.
    - As a preprocessing, it replaces `\$` to `&#x24;`, to skip escaped dollar character.
      This is because of roll back for it.
* Text `<pre><code class="language-katex"> ... </code></pre>`, `<span class="inlkatex"> ... </span>` are
  all rolled back to `$$ ... $$` and `$ ... $`, respectively.
    - Roll back for math expression replacement.

lambdalisue reviewed (changes requested):
I'm sorry but it's too complicated to review.

KSR-Yasuda commented:
@lambdalisue Oh, wrongly re-requested review. Ignore it for now.

KSR-Yasuda commented:
@lambdalisue

> I'm sorry but it's too complicated to review.

I admit it is not an appropriate way since it might break other Markdown syntax.
We should choose another if there's more fine way.

But, as I say in #190, math syntax `$ ... $` and `$$ ... $$` should be supported,
since current ` ```katex ... ``` ` code block would not work as math expressions in other Markdown viewers.

Does anybody have a good idea?

I also tried [Markdown-it KaTeX plugin](https://mdit-plugins.github.io/katex.html), but could not work it.
I suppose there may be KaTeX support like that, though, I haven't found one that matched with Previm.

KSR-Yasuda committed:
Add `{g,b}:previm_katex_inline_expr` option

KSR-Yasuda committed:
Fix false syntax application for `*` and `_` in inline katex expressions

KSR-Yasuda commented:
Oh, it had failed the cases like below:

```markdown
${x}_{i}$ aaa abc ${x}_{j}$ aaa
```

The `_` appearances in `$ inline katex expression $` had been considered as *`_emphasis_`*.

Fixed by escaping to HTML entity reference expressions.

KSR-Yasuda committed:
Fix for backslash appearance in inline katex expressions

- KaTeX expressions such as `\,` was not processed correctly in inline expressions.

KSR-Yasuda committed:
Fix failed replacement rollback

KSR-Yasuda committed:
Merge branch 'master' into feature/TexExpr
